src/Data/BitcoinNode.ts

```diff
@@ -165,17 +172,24 @@ export class BitcoinNodeManager {
     }
 
     async list_transactions(count: number): Promise<any> {
-        return (
-            await window.electron.bitcoin_command([
-                { method: 'listtransactions', parameters: ['*', count] },
-            ])
-        )[0];
+        const [result] = await window.electron.bitcoin_command([
+            { method: 'listtransactions', parameters: ['*', count] },
+        ]);
+        if (result.name === 'RpcError') {
+            throw result;
+        }
+        return result
     }
     async generate_blocks(n: number): Promise<void> {
         const addr = await this.get_new_address();
-        await window.electron.bitcoin_command([
-            { method: 'generatetoaddress', parameters: [10, addr] },
+        console.log(addr);
+        const [result] = await window.electron.bitcoin_command([
+            { method: 'generatetoaddress', parameters: [n, addr] },
         ]);
+        if (result.name === 'RpcError') {
```

Comment:
if result is a void is that safe?

Reply:
I've included checks if a call returns a void. In order to avoid doing those checks in every single method I've added an `execute` method that verifies the result. I've added the `executeBatch` just for consistency.